internal/services/notification/webhooks.go

```diff
@@ -115,3 +125,66 @@ func (n *NotificationService) generatewebhook(ctx context.Context, ev *rstypes.R
 
 	return webhook
 }
+
+func (n *NotificationService) runWebhooksCleanerLoop(ctx context.Context, webhookExpireInterval time.Duration) {
+	n.log.Debug().Msgf("webhookCleanerLoop")
+
+	for {
+		if err := n.runWebhooksCleaner(ctx, webhookExpireInterval); err != nil {
+			n.log.Warn().Err(err).Msgf("webhooksCleaner error")
+		}
+
+		sleepCh := time.NewTimer(webhookExpireInterval).C
+		select {
+		case <-ctx.Done():
+			return
+		case <-sleepCh:
+		}
+	}
+}
+
+func (n *NotificationService) runWebhooksCleaner(ctx context.Context, webhookExpireInterval time.Duration) error {
+	l := n.lf.NewLock(webhooksCleanerLockKey)
+	if err := l.TryLock(ctx); err != nil {
+		if errors.Is(err, lock.ErrLocked) {
+			return nil
+		}
+		return errors.WithStack(err)
+	}
+	defer func() { _ = l.Unlock() }()
+
+	for {
+		var runWebhooks []*nstypes.RunWebhook
+
+		err := n.d.Do(ctx, func(tx *sql.Tx) error {
+			var err error
+			runWebhooks, err = n.d.GetRunWebhooks(tx, maxRunWebhooksQueryLimit)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			for _, r := range runWebhooks {
+				err = n.d.DeleteRunWebhookDeliveriesByRunWebhookID(tx, r.ID)
```

Comment:
Are you deleting all the webhooks ignoring webhookExpireInterval or am I missing something?

If so there's something missing in the tests.